src/URI.cc

```diff
@@ -1000,7 +1036,8 @@ const URIFragment &URI::Fragment() const
 void URI::Clear()
 {
   this->dataPtr->scheme.clear();
-  this->dataPtr->authority.Clear();
+  if (this->dataPtr->authority)
+    (*this->dataPtr->authority).Clear();
```

Comment:
I believe it may make more sense to reset the authority here, which makes the optional state empty.
```suggestion
    this->dataPtr->authority.reset();
```

Reply:
1b7d94a8ddf046087c595734e8b7ae5dfd3b532c